### PR TITLE
[SPARK-44268][CORE][TEST][FOLLOWUP] Add test to generate `sql-error-conditions` doc automatic

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
This is a follow up PR for #41813, change test to automatic generate `sql-error-conditions` doc. Like other GOLDEN_FILES test, if the doc not match will report a error during test.

### Why are the changes needed?
1. make sure the `error-classes.json` sync with doc
2. and developer sync doc easier.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
add new test.

### CODE REVIEW

File: docs/sql-error-conditions-datatype-mismatch-error-class.md [hunk 1]
```diff
@@ -176,16 +174,13 @@ A range window frame cannot be used in an unordered window specification.
...
 1. The start and stop expressions must resolve to the same type.
-
 2. If start and stop expressions resolve to the `<startType>` type, then the step expression must resolve to the `<stepType>` type.
-
 3. Otherwise, if start and stop expressions resolve to the `<otherStartType>` type, then the step expression must resolve to the same type.
 
 ## SPECIFIED_WINDOW_FRAME_DIFF_TYPES
 
-Window frame bounds `<lower>` and `<upper>` do not have the same type: `<lowerType>` <> `<upperType>`.
+Window frame bounds `<lower>` and `<upper>` do not have the same type: `<lowerType>` `<>` `<upperType>`.
```
Comment (MaxGekk) [inline docs/sql-error-conditions-datatype-mismatch-error-class.md:183]: Should leave <> as is?

File: docs/sql-error-conditions-datatype-mismatch-error-class.md [hunk 2]
```diff
@@ -205,7 +200,7 @@ The lower bound of a window frame must be `<comparison>` to the upper bound.
 
 ## STACK_COLUMN_DIFF_TYPES
 
-The data type of the column (`<columnIndex>`) do not have the same type: `<leftType>` (`<leftParamIndex>`) <> `<rightType>` (`<rightParamIndex>`).
+The data type of the column (`<columnIndex>`) do not have the same type: `<leftType>` (`<leftParamIndex>`) `<>` `<rightType>` (`<rightParamIndex>`).
```
Comment (MaxGekk) [inline docs/sql-error-conditions-datatype-mismatch-error-class.md:203]: ditto: don't touch <>

File: docs/sql-error-conditions-incomplete-type-definition-error-class.md [hunk 3]
```diff
@@ -31,10 +31,10 @@ The definition of "ARRAY" type is incomplete. You must provide an element type.
 
 ## MAP
 
-The definition of "MAP" type is incomplete. You must provide a key type and a value type. For example: "MAP<TIMESTAMP, INT>".
+The definition of "MAP" type is incomplete. You must provide a key type and a value type. For example: "MAP`<TIMESTAMP, INT>`".
```
Comment (MaxGekk) [inline docs/sql-error-conditions-incomplete-type-definition-error-class.md:34]: Similar issue here, don't quote <TIMESTAMP, INT>

File: docs/sql-error-conditions-invalid-parameter-value-error-class.md [hunk 4]
```diff
@@ -51,6 +51,6 @@ Expects group index between 0 and `<groupCount>`, but got `<groupIndex>`.
 
 ## ZERO_INDEX
 
-expects `%1$`, `%2$` and so on, but got `%0$`.
+expects %1$, %2$ and so on, but got %0$.
```
Comment (MaxGekk) [inline docs/sql-error-conditions-invalid-parameter-value-error-class.md:54]: This one should have quotes, shouldn't it?

File: docs/sql-error-conditions-invalid-format-error-class.md [hunk 5]
```diff
@@ -55,7 +55,7 @@ Thousands separators (, or G) may not appear after the decimal point in the numb
 
 ## UNEXPECTED_TOKEN
 
-Found the unexpected `<token>` in the format string; the structure of the format string must match: `[MI|S]` `[$]` `[0|9|G|,]*` `[.|D]` `[0|9]*` `[$]` `[PR|MI|S]`.
+Found the unexpected `<token>` in the format string; the structure of the format string must match: [MI|S] [$] [0|9|G|,]* [.|D] [0|9]* [$] [PR|MI|S].
```
[the remark this answers is not in the document]
Comment (MaxGekk) [reply docs/sql-error-conditions-invalid-format-error-class.md:58]: Yep

**Review by MaxGekk — APPROVED** [review]

**Review by beliefer — APPROVED** [review]
LGTM.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**